design/lvm/resize_workflow.md

```diff
@@ -0,0 +1,151 @@
+
+### Steps to perform volume expansion
+
+1. Edit PVC and update capacity to desired value. In below example updated
+   the PVC capacity from 4Gi to 8Gi.
+
+```sh
+kubectl patch pvc csi-lvmpv -p '{"spec":{"resources":{"requests":{"storage":"8Gi"}}}}'
+persistentvolumeclaim/csi-lvmpv patched
+```
+
+2. User can observe the resize related events by describing PVC
+
+```sh
+kubectl describe pvc csi-lvmpv
+...
+...
+Events:
+  Type     Reason                      Age   From                                                                                Message
+  ----     ------                      ----  ----                                                                                -------
+  Normal   Provisioning                11m   local.csi.openebs.io_openebs-lvm-controller-0_b4700a50-b7cd-4de5-bc26-d3dd832ac9eb  External provisioner is provisioning volume for claim "default/csi-lvmpv"
+  Normal   ExternalProvisioning        11m   persistentvolume-controller                                                         waiting for a volume to be created, either by external provisioner "local.csi.openebs.io" or manually created by system administrator
+  Normal   ProvisioningSucceeded       11m   local.csi.openebs.io_openebs-lvm-controller-0_b4700a50-b7cd-4de5-bc26-d3dd832ac9eb  Successfully provisioned volume pvc-f532e80d-b39b-4801-837b-57a47ae08ea8
+  Normal   Resizing                    95s   external-resizer local.csi.openebs.io                                               External resizer is resizing volume pvc-f532e80d-b39b-4801-837b-57a47ae08ea8
+  Warning  ExternalExpanding           95s   volume_expand                                                                       Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
+  Normal   FileSystemResizeRequired    95s   external-resizer local.csi.openebs.io                                               Require file system resize of volume on nod
+```
+
+3. Once the filesystem expansion is succeeded then success events will
+   be generated and status of PVC also will be updated to latest capacity.
+```sh
+kubectl describe pvc csi-lvmpv
+
+ Normal   FileSystemResizeSuccessful  21s   kubelet                                                                             MountVolume.NodeExpandVolume succeeded for volume "pvc-f532e80d-b39b-4801-837b-57a47ae08ea8"
+```
+
+### High level Sequence Diagram
+
+Below is high level sequence diagram for volume expansion workflow
+
+![Volume Expansion Workflow](./images/resize_sequence_diagram.jpg)
+
+### Test Plan
+A test plan will include following test cases:
+- Test volume expansion operation on all supported filesystems(ext3, ext4, xfs, btrfs).
+- Test volume expansion while previous expansion of the volume is already in progress.
+- Restart of node LVM-Node-CSI-driver while filesystem expansion is in progress.
```

Comment:
test volume expansion with snapshot
test volume expansion for thick provisioned volume group (+ test case to over expand the volume that the vg szie)

Reply:
updated